fili-core/src/test/groovy/com/yahoo/bard/webservice/logging/LogFormatterProviderSpec.groovy

```diff
@@ -0,0 +1,67 @@
+package com.yahoo.bard.webservice.logging
+
+import com.yahoo.bard.webservice.config.SystemConfig
+import com.yahoo.bard.webservice.config.SystemConfigException
+import com.yahoo.bard.webservice.config.SystemConfigProvider
+
+import spock.lang.Shared
+import spock.lang.Specification
+
+
+class LogFormatterProviderSpec extends Specification {
+
+    @Shared LogFormatter originalLogFormatter
+    String originalLogFormatterClassName
+    SystemConfig systemConfig = SystemConfigProvider.instance;
+    String logFormatterKey = systemConfig.getPackageVariableName(
+            LogFormatterProvider.LOG_FORMATTER_IMPLEMENTATION_SETTING_NAME
+    )
+
+    def setupSpec() {
+        originalLogFormatter = LogFormatterProvider.logFormatter
+    }
+
+    def cleanupSpec() {
+        LogFormatterProvider.logFormatter = originalLogFormatter
+    }
+
+    def setup() {
+        LogFormatterProvider.logFormatter = null
+        try {
+            originalLogFormatterClassName = systemConfig.getStringProperty(logFormatterKey)
+        } catch(SystemConfigException ignored) {
+            originalLogFormatterClassName = null
+        }
+    }
+
+    def cleanup() {
+        if (originalLogFormatterClassName == null) {
+            systemConfig.clearProperty(logFormatterKey)
+        } else {
+            systemConfig.setProperty(logFormatterKey, originalLogFormatterClassName)
+        }
+    }
+
+    def "The LogFormatterProvider instantiates the configured logFormatter instance"() {
+        given: "We haven't instantiated the log formatter yet"
+        assert LogFormatterProvider.logFormatter == null
```

Comment:
The reason I'm using a `given` and not an `expect` is because I'm asserting a precondition rather than performing any actual tests. To me, when I see an `expect` block I see "here be tests of this feature" not "here be preconditions."

Also, "given" is often used when specifying formal preconditions in english, (well the english used by mathematicians) so this reads very naturally as "given the logFormatter is null."

I agree that the `assert` is annoying, but I think this more clearly calls out that this is merely asserting a precondition for the test, and not actually testing anything.